sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -41,186 +42,433 @@ enum EntityType {
+    describe(`Atom management - List on "${entityType}" entities #RunInBrowser`, function(): void {
+      beforeEach(async () => {
+        switch (entityType) {
+          case EntityType.QUEUE:
+            await recreateQueue(managementQueue1);
+            await recreateQueue(managementQueue2);
+            break;
+
+          case EntityType.TOPIC:
+            await recreateTopic(managementTopic1);
+            await recreateTopic(managementTopic2);
+            break;
+
+          case EntityType.SUBSCRIPTION:
+            await recreateTopic(managementTopic1);
+            await recreateSubscription(managementTopic1, managementSubscription1);
+            await recreateSubscription(managementTopic1, managementSubscription2);
+            break;
+
+          case EntityType.RULE:
+            await recreateTopic(managementTopic1);
+            await recreateSubscription(managementTopic1, managementSubscription1);
+            await createEntity(
+              EntityType.RULE,
+              managementRule1,
+              managementTopic1,
+              managementSubscription1
+            );
+            await createEntity(
+              EntityType.RULE,
+              managementRule2,
+              managementTopic1,
+              managementSubscription1
+            );
+            break;
+          default:
+            throw new Error("TestError: Unrecognized EntityType");
+        }
+      });
+
+      afterEach(async () => {
+        switch (entityType) {
+          case EntityType.QUEUE:
+            await deleteEntity(EntityType.QUEUE, managementQueue1);
+            await deleteEntity(EntityType.QUEUE, managementQueue2);
+            break;
+
+          case EntityType.TOPIC:
+            await deleteEntity(EntityType.TOPIC, managementTopic1);
+            await deleteEntity(EntityType.TOPIC, managementTopic2);
+            break;
+          case EntityType.SUBSCRIPTION:
+          case EntityType.RULE:
+            await deleteEntity(EntityType.TOPIC, managementTopic1);
+            break;
+
+          default:
+            throw new Error("TestError: Unrecognized EntityType");
+        }
+      });
+
+      it(`List on existing entities for type ${entityType} with top 1 returns the first entity`, async () => {
+        const allEntities = await listEntities(
+          entityType,
+          managementTopic1,
+          managementSubscription1
+        );
+
+        const topOneEntity = await listEntities(
+          entityType,
+          managementTopic1,
+          managementSubscription1,
+          undefined,
+          1
+        );
+
+        should.equal(
+          Array.isArray(topOneEntity),
+          true,
+          "Result must be any array for list requests"
+        );
+        should.equal(topOneEntity.length, 1, "Result must be an empty array");
+        should.equal(
+          allEntities[0][entityType.toLowerCase() + "Name"],
+          topOneEntity[0][entityType.toLowerCase() + "Name"],
```

Comment:
Is the order of the entities in the result of a list operation guaranteed to be the same? If not, then the check above can fail.

Reply:
Agreed, removed the check